x/posts/keeper/msg_server_post_owner_transfer.go

```diff
@@ -0,0 +1,218 @@
+	"github.com/desmos-labs/desmos/v5/x/posts/types"
+)
+
+// RequestPostOwnerTransfer defines the rpc method for Msg/RequestPostOwnerTransfer
+func (k msgServer) RequestPostOwnerTransfer(goCtx context.Context, msg *types.MsgRequestPostOwnerTransfer) (*types.MsgRequestPostOwnerTransferResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	// Check if the sender has profile
+	if !k.HasProfile(ctx, msg.Sender) {
+		return nil, errors.Wrapf(sdkerrors.ErrInvalidRequest, "you cannot transfer a post without having a profile")
+	}
+
+	// Check if the receiver has profile
+	if !k.HasProfile(ctx, msg.Receiver) {
+		return nil, errors.Wrapf(sdkerrors.ErrInvalidRequest, "you cannot transfer a post to the user having no profile")
+	}
+
+	// Make sure the receiver has not blocked the sender
+	if k.HasUserBlocked(ctx, msg.Receiver, msg.Sender, msg.SubspaceID) {
+		return nil, errors.Wrapf(sdkerrors.ErrInvalidRequest, "receiver has blocked you")
+	}
+
+	// Get the post
+	post, found := k.GetPost(ctx, msg.SubspaceID, msg.PostID)
+	if !found {
+		return nil, errors.Wrapf(sdkerrors.ErrInvalidRequest, "post with id %d does not exist", msg.PostID)
+	}
+
+	// Make sure the sender matches the owner
+	if post.Owner != msg.Sender {
+		return nil, errors.Wrapf(sdkerrors.ErrInvalidRequest, "you cannot transfer a post that you do not own")
+	}
+
+	// Check if the post owner transfer request exists
+	if k.HasPostOwnerTransferRequest(ctx, msg.SubspaceID, msg.PostID) {
+		return nil, errors.Wrap(sdkerrors.ErrInvalidRequest, "post owner transfer request already exists")
+	}
+
+	// Save the post owner transfer request
+	k.SavePostOwnerTransferRequest(ctx, types.NewPostOwnerTransferRequest(
+		msg.SubspaceID,
+		msg.PostID,
+		msg.Receiver,
+		msg.Sender,
+	))
+
+	ctx.EventManager().EmitEvents(sdk.Events{
+		sdk.NewEvent(
+			sdk.EventTypeMessage,
+			sdk.NewAttribute(sdk.AttributeKeyModule, types.AttributeValueCategory),
+			sdk.NewAttribute(sdk.AttributeKeyAction, sdk.MsgTypeURL(msg)),
+			sdk.NewAttribute(sdk.AttributeKeySender, msg.Sender),
+		),
+		sdk.NewEvent(
+			types.EventTypeRequestPostOwnerTransfer,
+			sdk.NewAttribute(types.AttributeKeySubspaceID, fmt.Sprintf("%d", msg.SubspaceID)),
+			sdk.NewAttribute(types.AttributeKeyPostID, fmt.Sprintf("%d", msg.PostID)),
+			sdk.NewAttribute(types.AttributeKeyReceiver, msg.Receiver),
+			sdk.NewAttribute(types.AttributeKeySender, msg.Sender),
+		),
+	})
+
+	return &types.MsgRequestPostOwnerTransferResponse{}, nil
+}
+
+// CancelPostOwnerTransfer defines the rpc method for Msg/CancelPostOwnerTransfer
+func (k msgServer) CancelPostOwnerTransfer(goCtx context.Context, msg *types.MsgCancelPostOwnerTransfer) (*types.MsgCancelPostOwnerTransferResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	// Check if the sender has profile
+	if !k.HasProfile(ctx, msg.Sender) {
+		return nil, errors.Wrapf(sdkerrors.ErrInvalidRequest, "you cannot cancel a post owner transfer request without having a profile")
+	}
```

Comment:
Is this wanted? If the request was created, it means that the user had the profile at that time. I think we should allow them to cancel the request even if they have deleted them in the meanwhile

Reply:
Yeah, we should allow all the users to cancel/refuse requests. Updated.